terraform/.gitignore

```diff
@@ -0,0 +1,6 @@
+.idea/
+.terraform/
+.terraform.lock.hcl
+venv/
+**/*.tfstate
+**/*.tfstate.backup
```

Comment:
I reckon we can put these in the `.gitignore` in the root of the project, I think it keeps things simple if there's only one `.gitignore`.

Reply:
Isn't it simpler to have a separate `.gitignore` here, specifically for the files in this folder?

Reply:
I think it can then be surprising, for example if a file is ignored and you're not sure why, you have to check all the `.gitignore` files to figure out why.